sdlib/d/gc/tcache.d

```diff
@@ -98,12 +98,14 @@ public:
 			}
 		} else {
 			auto esize = pd.extent.size;
-			if (samePointerness && alignUp(size, PageSize) == esize) {
+			if (samePointerness && (alignUp(size, PageSize) == esize
+				    || (isLargeSize(size) && pd
+					    .arena
+					    .resizeLarge(emap, pd.extent, getAllocSize(size))))) {
```

Comment:
I don't think we necessarily want to use getAllocSize here.

Reply:
I think we expects this for extend, but I'm not sure about realloc. In any case, this is not a blocker.

Reply:
Sleeping on this, this actually doesn't make sense. By pessimising the size, we actually increase our chances of the operation failing, and therefore requiring a copy. This is strictly worse.